### PR TITLE
allow passing arbitrary kwargs to requests

### PR DESCRIPTION
**Pull request by evgeni**
this e.g. allows to do file uploads by passing `files=…` to call()

### CODE REVIEW

File: apypie/api.py [hunk 1]
```diff
@@ -156,22 +156,20 @@ def call(self, resource_name, action_name, params={}, headers={}, options={}):
...
-            headers, options)
+            headers, options, **kwargs)
 
-    def http_call(self, http_method, path, params=None, headers=None, options=None):
+    def http_call(self, http_method, path, params=None, headers=None, options=None, **kwargs):
         full_path = urljoin(self.uri, path)
-        kwargs = {
-            'verify': self._session.verify,
-        }
+        kwargs.update({'verify': self._session.verify})
```
Comment (ekohl) [inline apypie/api.py]: Why do we do this? We already set `verify` on the session so that sounds redundant.

File: apypie/api.py [hunk 2]
```diff
@@ -156,22 +156,20 @@ def call(self, resource_name, action_name, params={}, headers={}, options={}):
         if not options.get('skip_validation', False):
             action.validate(params)
 
-        return self._call_action(action, params, headers, options)
+        return self._call_action(action, params, headers, options, **kwargs)
 
-    def _call_action(self, action, params={}, headers={}, options={}):
+    def _call_action(self, action, params={}, headers={}, options={}, **kwargs):
```
Comment (ekohl) [inline apypie/api.py]: On a side note: `options` is unused

File: apypie/api.py (hunk 1 above)
Comment (evgeni) [reply apypie/api.py]: https://github.com/Apipie/apypie/pull/25

File: apypie/api.py (hunk 2 above)
Comment (evgeni) [reply apypie/api.py]: I noticed that too, yeah, but didn't want to creep the PR.

**Comment by evgeni** [comment]
> How many more parameters are there? Should we just expose `files` as a parameter?

We could. Today `files` is all I need, but who knows tomorrow :)

File: apypie/api.py (hunk 1 above)
Comment (ekohl) [reply apypie/api.py]: Perhaps good to add a comment about this :) I'm also wondering if we should update kwargs or instantiate a new dict and update that.

**Comment by ekohl** [comment]
> We could. Today `files` is all I need, but who knows tomorrow :)

I prefer to avoid kwargs in a formal API if I can. Makes it easier to properly document and test.

**Pull request merged**